### DISCUSSION TITLE
Cursor style not respected on inactive window

### DISCUSSION THREAD

**Discussion by SenseiMarv** [ghostty-org/ghostty#3750] [Feature Requests, Ideas]
When the Ghostty window is inactive, the cursor will show the block style, even if a different style is set in the config:

![image](https://github.com/user-attachments/assets/c7f6e6f5-3a3d-4da0-973c-379e833d16f9)

In iterm2, the bar will also be displayed when the window is inactive, so I wonder whether this behavior should be adapted.

**Comment by sionescu** [comment]
I can reproduce this on Linux: I set `cursor-style=underline` but it changes to a bar in inactive windows.

**Comment by mitchellh** [comment]
This is currently working as designed and is not a bug. I believe someone suggested a `cursor-style-unfocused` config (that could be set to a value that kept it to whatever it was).

**Reply by SenseiMarv** [reply]
If the current behaviour is intended, I think it would make sense to have such a config option so that people who want a consistent cursor across different window states can control it themselves. Should an issue be created to add the config then?

**Reply by mitchellh** [reply]
I think a PR already exists for this that’s unreviewed

**Reply by SenseiMarv** [reply]
Good to know, I had a look and this is it: https://github.com/ghostty-org/ghostty/pull/2818

It seems that the contributor didn't have the chance yet to assess your suggestions. Since there is no issue related to the PR, if it stays untouched and is being closed at some point due to that, there would be no issue to track the feature. If you're fine with that, I can close this discussion, otherwise an issue would probably still make sense.
